src/components/SearchInput/SearchInput.js

```diff
@@ -14,32 +14,46 @@
  */
 
 import React from 'react';
+import PropTypes from 'prop-types';
+
+import styled from '@emotion/styled';
 
 import { childrenPropType } from '../../util/shared-prop-types';
 
 import Input from '../Input';
-import { ReactComponent as SearchIcon } from './search.svg';
+import { ReactComponent as SearchIcon } from './icons/search.svg';
+import { ReactComponent as ClearIcon } from './icons/clear.svg';
+
+const StyledClearIcon = styled(ClearIcon)`
+  pointer-events: all !important;
+  cursor: pointer !important;
+`;
 
 /**
  * SearchInput component for forms.
  */
-const SearchInput = ({ children, ...props }) => (
+const SearchInput = ({ children, value, onClear, ...props }) => (
   <Input
-    {...props}
-    type="search"
+    value={value}
     renderPrefix={({ className }) => <SearchIcon {...{ className }} />}
+    renderSuffix={({ className }) =>
+      value ? <StyledClearIcon onClick={onClear} {...{ className }} /> : null
+    }
```

Comment:
What do you think about only adding the clear button if the `onClear` prop is defined?

Reply:
Makes total sense! Will do it ;)